src/query/query.pest

```diff
@@ -15,12 +15,14 @@ expr = _{ prefix? ~ (brackets | singleVariableFunc | noVariableFunc | infixExpre
 
 brackets = { "(" ~ body ~ ")" }
 
-noVariableFunc     =  { funcNames ~ "()" }
-singleVariableFunc =  { funcNames ~ "(" ~ body ~ ")" }
-funcNames          = _{ latest }
-latest             =  { "latest" }
-
-infixExpression = { lookup ~ infixFunction ~ lookupValue }
+noVariableFunc          =  { noVariableFuncNames ~ "()" }
+singleVariableFunc      =  { singleVariableFuncNames ~ "(" ~ body ~ ")" }
+noVariableFuncNames     =  _{ latest }
+singleVariableFuncNames =  _{ latest | single }
+latest                  =  { "latest" }
+single                  =  { "single" }
+
+infixExpression = { lookup ~ infixFunction ~ lookupValue | lookupValue ~ infixFunction ~ lookup }
```

Comment:
should this be

```suggestion
infixExpression = { (lookup | lookupValue)  ~ infixFunction ~  (lookup | lookupValue)}
```

(I may have the grouping operator wrong here!). So even though it makes no sense a user could write

```
latest(1 == 2)
```

which returns nothing useful ever, or

```
latest(parameter:a == parameter:b)
```

which is occasionally useful

Reply:
Hmm I could make that do able but have explicitly made it impossible (see tests here https://github.com/mrc-ide/outpack_server/pull/41/files#diff-41c5626f836cdc731bc0ed84cf86571988982366fa33facc8354a03f4e9660d6R473-R484)

Suppose that will just add a couple more cases to check in the evaluation - will take a look at that

Reply:
One please that this does turn up nontrivially, but hard to do much with in the rust version is `parameter:a == this:a` or `parameter:a == envir:a` which we use in quite a few places fetching dependencies in the R version - if we use this as the basis of the python support we will need that

Reply:
Ok I've given that a go now, you can compare different parameters and hopefully should be done in a way that we can add `envir` and `this` types in the future when we need them